wicket/src/state/update.rs

```diff
@@ -21,12 +21,14 @@ use std::collections::BTreeMap;
 use std::fmt::Display;
 use wicketd_client::types::{ArtifactId, SemverVersion};
 
+type ArtifactVersions = Vec<(SemverVersion, Option<Vec<u8>>)>;
```

Comment:
Should this be a map instead of a vec? Is it okay to have multiple identical versions with potentially-different `Option<Vec<u8>>` values?

Reply:
I was trying to map the existing behavior of the map of artifact kind to one version to a map of artifact kind to a vec of all possible versions. I think this is the same issue with using a tuple instead of a named struct. I'll see if that makes it more obvious what's going on here.

Reply:
I kept the same `Vec` concept but made it a struct to be a bit more obvious what's going on. The same version with different `Option<Vec<u8>>` matches the behavior we have now for the RoT: the same version with different `SIGN` values and we select based on the `SIGN` value.